operator/v1/types_console.go

```diff
@@ -202,6 +205,52 @@ type AddPage struct {
 	DisabledActions []string `json:"disabledActions,omitempty"`
 }
 
+type PerspectiveState string
+
+// These are valid perspective states. "Enabled" means the perspective is shown.
+// "Disabled" means the Perspective is hidden.
+// "AccessReview" means access review check is required to show or hide a Perspective.
+const (
+	Enabled      PerspectiveState = "Enabled"
+	Disabled     PerspectiveState = "Disabled"
+	AccessReview PerspectiveState = "AccessReview"
+)
+
+// `required` and  `missing` can work together esp. in the case where cluster admin
+// wants to show another perspective to users without specific permissions. If both `required` and `missing`
+// are not specified for `state: AccessReview`, access review check will be ignored and perspective will be enabled.
+type PerspectiveAccessReview struct {
+	// required defines a list of permission checks. The perspective will only be shown when all checks are successful. If no checks are specified then we simply ignore the check even if the state is AccessReview.
+	// +optional
+	Required []authorizationv1.ResourceAttributes `json:"required,omitempty"`
+	// missing defines a list of permission checks. The perspective will only be shown when at least one check fails. If no checks are specified then we simply ignore the check even if the state is AccessReview.
+	// +optional
+	Missing []authorizationv1.ResourceAttributes `json:"missing,omitempty"`
```

Comment:
I don't think the current description describes what the end user sees, is it more clear with this suggestion? I'm trying to explain it in terms of what happens to the perspective
```suggestion
	// required defines a list of permission checks. The perspective will only be shown when all checks are successful. When omitted, the access review is skipped and the perspective will not be shown unless it is required to do so based on the configuration of the missing access review list.
	// +optional
	Required []authorizationv1.ResourceAttributes `json:"required,omitempty"`
	// missing defines a list of permission checks. The perspective will only be shown when at least one check fails. When omitted, the access review is skipped and the perspective will not be shown unless it is required to do so based on the configuration of the required access review list.
	// +optional
	Missing []authorizationv1.ResourceAttributes `json:"missing,omitempty"`
```